packages/nx-php-e2e/src/nx-php.spec.ts

```diff
@@ -139,16 +139,153 @@ describe('nx-php', () => {
+      
+      // Add dependency to the library (using the same naming pattern that would be generated)
+      appComposer.require = appComposer.require || {};
+      appComposer.require[libName] = '^1.0.0';
+      
+      writeFileSync(appComposerPath, JSON.stringify(appComposer, null, 2));
+
+      // Reset the project graph to ensure changes are picked up
+      execSync(`npx nx reset`, {
+        cwd: projectDirectory,
+        stdio: 'inherit',
+      });
+
+      // Verify that both projects are still recognized and the app shows a dependency
+      const showAppOutput = execSync(`npx nx show project ${appName}`, {
+        cwd: projectDirectory,
+        encoding: 'utf-8',
+      });
+      
+      const showLibOutput = execSync(`npx nx show project ${libName}`, {
         cwd: projectDirectory,
         encoding: 'utf-8',
       });
-      expect(listOutput).toContain(projectName);
+
+      // Both projects should be discoverable
+      expect(showAppOutput).toContain(appName);
+      expect(showLibOutput).toContain(libName);
+      
+      // Verify project types
+      expect(showAppOutput).toContain('library'); // Generated projects default to library
+      expect(showLibOutput).toContain('library');
+      
+    }, 45000);
+
+    it('should correctly identify affected projects based on dependencies', () => {
+      const libName = 'shared-lib';
+      const appName = 'main-app';
+
+      // Generate a library project
+      execSync(`npx nx g @nx-php/composer:project ${libName}`, {
+        cwd: projectDirectory,
+        stdio: 'inherit',
+      });
+
+      // Generate an application project
+      execSync(`npx nx g @nx-php/composer:project ${appName}`, {
+        cwd: projectDirectory,
+        stdio: 'inherit',
+      });
+
+      // Modify the app's composer.json to depend on the library
+      const appComposerPath = join(projectDirectory, 'packages', appName, 'composer.json');
+      const appComposer = JSON.parse(readFileSync(appComposerPath, 'utf-8'));
+      
+      // Add dependency to the library
+      appComposer.require = appComposer.require || {};
+      appComposer.require[libName] = '^1.0.0';
+      
+      writeFileSync(appComposerPath, JSON.stringify(appComposer, null, 2));
+
+      // Create some files to test with
+      const libSourcePath = join(projectDirectory, 'packages', libName, 'src');
+      const appSourcePath = join(projectDirectory, 'packages', appName, 'src');
+      
+      // Create directories if they don't exist
+      execSync(`mkdir -p ${libSourcePath}`, { cwd: projectDirectory });
+      execSync(`mkdir -p ${appSourcePath}`, { cwd: projectDirectory });
+      
+      // Create source files
+      const libFilePath = join(libSourcePath, 'Library.php');
+      const appFilePath = join(appSourcePath, 'App.php');
+      
+      writeFileSync(libFilePath, '<?php\nclass Library {}\n');
+      writeFileSync(appFilePath, '<?php\nclass App {}\n');
```

Comment:
Nitpick, you don't need to actually write files to the filesystem for the affected command to run successfully.